_posts/2024-12-26-Reducing-false-positives-through-algorithmic-improvements.md

```diff
@@ -0,0 +1,264 @@
+
+#### RDS CPU utilization (rds_cpu_utilization_cc0c53)
+
+- **Dataset**: NAB AWS `rds_cpu_utilization_cc0c53`  
+- **Total records**: 4,031  
+- **Labeled anomalies**:  
+  - 2014-02-25 07:15:00 (Detected)  
+  - 2014-02-27 00:50:00 (Undetected)  
+- **Total RCF anomalies**: 1  
+
+![rds_cc0c53](/assets/media/blog-images/2024-12-26-Reducing-false-positives-through-algorithmic-improvements/rds_cc.png){:class="img-centered"}
+
+## Precision and recall
+
+Before we present the results, let's define two key concepts in machine learning evaluation:
+
+- **Precision**: The proportion of detected anomalies that are correct: of all the anomalies the algorithm detected, how many were true anomalies? High precision means few mistaken identifications. For example, if the algorithm detects 10 anomalies and 8 are actually anomalies, then the precision is $$8 / 10 = 0.8$$, or $$80%$$.  
+- **Recall**: The proportion of true anomalies that are detected: of all true anomalies, how many did the algorithm detect? High recall means few missed anomalies. For example, if there are 10 true anomalies and the algorithm detects 8 of them, then the recall is $$8 / 10 = 0.8$$, or $$80%$$.  
+
+### Results
+
+The following table summarizes RCF's precision and recall performance using the NAB CloudWatch benchmarks.
+
+| Dataset | Precision | Recall |
+|-----|-----------|--------|
+| AWS `ec2_cpu_utilization_24ae8d`  | 0.67      | 1      |
+| AWS `ec2_network_in_257a54`       | 0.25      | 1      |
+| AWS `ec2_disk_write_bytes_1ef3de` | 0.33      | 1      |
+| AWS `rds_cpu_utilization_e47b3b`  | 1         | 1      |
+| AWS `rds_cpu_utilization_cc0c53`  | 1         | 0.5    |
+
+Overall, RCF demonstrated strong recall, correctly detecting 7 out of 8 anomalies across the datasets. Notably, it achieved perfect recall in four of the five datasets. 
+
+RCF also maintains high precision. Most false positives occur early in the time series, before the model has accumulated sufficient historical data for accurate predictions. For example, in the following graph, the anomaly detected around April 12 at 3:14 for the `ec2_network_in_257a54` dataset is particularly noteworthy. It deviates from the previously observed pattern of uniform double spikes. Earlier patterns show the second spike declining within approximately five minutes, whereas this anomalous spike exhibits an extended decline lasting around ten minutes. 
+
+![network_zoom_in](/assets/media/blog-images/2024-12-26-Reducing-false-positives-through-algorithmic-improvements/network_zoom_in_1.png){:class="img-centered"}
+When excluding the first 20% of data as a probation period, precision improves across datasets. This adjustment highlights how RCF becomes more accurate after observing enough historical data. The results of the last 80% of the dataset are summarized in the following table.
+
```

Comment:
Line 203, first sentence: "of the data" => "of the dataset" (to match the last sentence)?